conans/client/loader.py

```diff
@@ -53,7 +53,10 @@ def load_class(self, conanfile_path):
 
     def load_export(self, conanfile_path, name, version, user, channel):
         conanfile = self.load_class(conanfile_path)
-
+        conanfile = conanfile(self._output, self._runner, user, channel)
+        if hasattr(conanfile, "preexport") and callable(conanfile.preexport):
+            conanfile.recipe_folder = os.path.dirname(conanfile_path)
```

Comment:
not sure about this new attribute... might be confusing like `conafile_folder` was.

Reply:
Can remove it and let users use ``__file__``.